### PR TITLE
fix(learn): remove white space for quick fix

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Related to: #39925 (does not close)

### CODE REVIEW

**Comment by Sky020** [comment]
Sorry for pushing this branch to the wrong repo. Should I close it and delete?

**Comment by nhcarrigan** [comment]
I think it should be okay. 🤔

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-higher-order-functions-map-filter-or-reduce-to-solve-a-complex-problem.md [hunk 1]
```diff
@@ -1,49 +1,53 @@
...
 ## Tests
+
 <section id='tests'>
 
 ```yml
 tests:
   - text: <code>squareList</code> should be a <code>function</code>.
     testString: assert.typeOf(squareList, 'function'), '<code>squareList</code> should be a <code>function</code>';
-  - text: for or while loops or forEach should not be used.
+  - text: <code>for</code> or <code>while</code> loops or <code>forEach</code> should not be used.
```
Comment (nhcarrigan) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-higher-order-functions-map-filter-or-reduce-to-solve-a-complex-problem.md:29]:
```suggestion
  - text: <code>for</code>, <code>while</code>, and <code>forEach</code> should not be used.
```

The wording felt a little strange here. How is this?

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
One tiny nitpick, otherwise LGTM!

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-higher-order-functions-map-filter-or-reduce-to-solve-a-complex-problem.md (hunk 1 above)
Comment (Sky020) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-higher-order-functions-map-filter-or-reduce-to-solve-a-complex-problem.md:29]: I thought it uncomfortable too. I was considering copying the note in the instructions:
```
Your function should not use any kind of <code>for</code> or <code>while</code> loops or the <code>forEach()</code> function.
```

Comment (nhcarrigan) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-higher-order-functions-map-filter-or-reduce-to-solve-a-complex-problem.md:29]: I think that's better, though still a little clunky. 

I'm almost tempted to say split it into two tests:
- `for` and `while` loops should not be used
- the `forEach()` method should not be used

Comment (Sky020) [reply curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-higher-order-functions-map-filter-or-reduce-to-solve-a-complex-problem.md:29]: I would much prefer **not** splitting it; the test is so tiny, and would bloat it.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-higher-order-functions-map-filter-or-reduce-to-solve-a-complex-problem.md [hunk 2]
```diff
@@ -1,49 +1,53 @@
...
     testString: assert(!__helpers.removeJSComments(code).match(/for|while|forEach/g));
   - text: <code>map</code>, <code>filter</code>, or <code>reduce</code> should be used.
-    testString: assert(__helpers.removeJSComments(code).match(/\.(map|filter|reduce)\s*\(/g));
+    testString: assert(__helpers.removeWhiteSpace(__helpers.removeJSComments(code)).match(/\.(map|filter|reduce)\(/g));
```
Comment (RandellDawson) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-higher-order-functions-map-filter-or-reduce-to-solve-a-complex-problem.md:32]: If you are just looking for the words `map`, `filter`, or `reduce`, I do not see the reason to remove the whitespace.

**Review by naomi-lgbt — APPROVED** [review]
LGTM~! 🥳 

Thank you for your pull request! We look forward to additional contributions!

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Pull request merged**